src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,222 @@
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        if (index == size - 1) {
+            return last.item;
+        }
+        iterationFromTheFirstElement(index);
+        iterationFromTheLastElement(index);
+        return node.item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> returnNode = null;
+        if (index == 0) {
+            node = new Node<>(null, value, first.next);
+            returnNode = first;
+            first.next.prev = node;
+            first = node;
+            return returnNode.item;
+        }
+        if (size - index > size / 2) {
+            returnNode = first;
+            for (int i = 1; i <= index; i++) {
+                returnNode = returnNode.next;
+            }
+        } else {
+            node = last;
+            for (int i = size - 1; i >= index; i--) {
+                returnNode = returnNode.prev;
+            }
+        }
+        node = new Node<>(returnNode.prev, value, returnNode.next);
+        returnNode.prev.next = node;
+        returnNode.next.prev = node;
+        return returnNode.item;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> removedNode = null;
+        if (index == 0) {
+            node = first;
+            if (size == 1) {
+                first = last = null;
+                size--;
+                return node.item;
+            }
+            first.next.prev = null;
+            first = first.next;
+            size--;
+            return node.item;
+        }
+        if (index == size - 1) {
+            node = last;
+            last = last.prev;
+            last.next = null;
+            size--;
+            return node.item;
+        }
+        iterationFromTheFirstElement(index);
+        iterationFromTheLastElement(index);
+        node.prev.next = node.next;
+        node.next.prev = node.prev;
+        size--;
+        return node.item;
     }
 
     @Override
     public boolean remove(T t) {
+
+        if (checkEquals(first, t)) {
+            if (size == 1) {
+                first = last = null;
+                size--;
+                return true;
+            }
+            first.next.prev = null;
+            first = first.next;
+            size--;
+            return true;
+        }
+        node = first;
+        for (int i = 0; i < size - 1; i++) {
+            node = node.next;
+            if (checkEquals(node, t)) {
+                node.prev.next = node.next;
+                node.next.prev = node.prev;
+                size--;
+                return true;
+            }
+            if (i == size - 1 && checkEquals(last, t)) {
+                node = last;
+                last = last.prev;
+                last.next = null;
+                size--;
+                return true;
+            }
+        }
```

Comment:
this method it's too large. Let's divide this to a few

Reply:
done